identity/identity.go

```diff
@@ -73,6 +74,39 @@ func Get(ctx context.Context) XRHID {
 	return ctx.Value(Key).(XRHID)
 }
 
+func (j *XRHID) checkHeader(w http.ResponseWriter) error {
+
```

Comment:
How about checking for 'Associate' first and returning nil striaght away?
Then you could check each other error and return a descriptive error.